src/commands/swapSlots.ts

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// tslint:disable-next-line:no-require-imports
+import WebSiteManagementClient = require('azure-arm-website');
+import { OutputChannel } from 'vscode';
+import { SiteClient } from 'vscode-azureappservice';
+import { AzureTreeDataProvider, IAzureNode, IAzureQuickPickItem } from 'vscode-azureextensionui';
+import { DeploymentSlotTreeItem } from '../explorer/DeploymentSlotTreeItem';
+import { nodeUtils } from '../utils/nodeUtils';
+
+export async function swapSlots(tree: AzureTreeDataProvider, sourceSlotNode: IAzureNode<DeploymentSlotTreeItem> | undefined, outputChannel: OutputChannel): Promise<void> {
+    if (!sourceSlotNode) {
+        sourceSlotNode = <IAzureNode<DeploymentSlotTreeItem>>await tree.showNodePicker(DeploymentSlotTreeItem.contextValue);
+    }
+    const sourceSlotClient: SiteClient = sourceSlotNode.treeItem.client;
+
+    const productionSlotLabel: string = 'production';
+    const deploymentSlots: IAzureNode<DeploymentSlotTreeItem>[] = <IAzureNode<DeploymentSlotTreeItem>[]>await sourceSlotNode.parent.getCachedChildren();
+    const otherSlots: IAzureQuickPickItem<DeploymentSlotTreeItem | undefined>[] = [{
+        label: productionSlotLabel,
+        description: 'Swap slot with production',
+        detail: '',
+        data: undefined
+    }];
+
+    for (const slot of deploymentSlots) {
+        if (sourceSlotClient.slotName !== slot.treeItem.client.slotName) {
+            // Deployment slots must have an unique name
+            const otherSlot: IAzureQuickPickItem<DeploymentSlotTreeItem | undefined> = {
+                label: slot.treeItem.client.slotName,
+                description: '',
+                data: slot.treeItem
+            };
+
+            otherSlots.push(otherSlot);
+        }
+    }
+
+    const quickPickOptions = { placeHolder: `"${sourceSlotClient.slotName}" will be swapped with the destination slot.`, ignoreFocusOut: true };

```

Comment:
This always felt confusing to me.  Maybe something along the lines of "Select which slot to swap with {slotname}"?